api.go

```diff
@@ -102,6 +103,40 @@ func (a *API) NumSlots() (int, error) {
 	return n, unmarshalPyON(s, &n)
 }
 
+// Uptime returns FAH uptime
+func (a *API) Uptime() (time.Duration, error) {
+	// work around for response to uptime command not including a trailing newline
+	s, err := a.Exec("eval \"$(uptime)\\n\"")
```

Comment:
I think there should be a helper function for `Exec()` for dealing with these commands which don't end in newline. It should `eval` the sub-command like what you did. (I call it a sub-command because it's probably used for inlining.)

At first I was thinking of doing this:

```diff
diff --git a/api.go b/api.go
index 198fbfd..801f3d6 100644
--- a/api.go
+++ b/api.go
@@ -252,9 +252,9 @@ func readMessage(r telnet.Reader) (string, error) {
 
 		buffer.WriteByte(b[0])
 
-		const endOfMessage = "\n> "
+		const endOfMessage = "> "
 		if strings.HasSuffix(buffer.String(), endOfMessage) {
-			return strings.TrimPrefix(strings.TrimSuffix(buffer.String(), "\n> "), "\n"), nil
+			return strings.TrimPrefix(strings.TrimSuffix(buffer.String(), "> "), "\n"), nil
 		}
 	}
 }

```

But then that might cause issues with embedded `"> "` in the middle of a message.

Instead of `"eval \"$(uptime)\\n\""`, you can use a string literal like this `` `eval "$(info)\n"` `` to make it easier to read.

Reply:
Yes, I came to the same conclusion regarding embedded `>` in the response, which is how I came to the `eval` hack.

A helper is a good idea. Any suggestions for its name?

Reply:
What about

```go
// ExecEval can execute commands which do not return a trailing newline.
func (a *API) ExecEval(command string) (string, error) {
```